server/src/main/java/com/thoughtworks/go/server/newsecurity/filterchain/AuthorizeAgentFilterChain.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.newsecurity.filterchain;
+
+import com.thoughtworks.go.server.newsecurity.filters.DenyAllAccessFilter;
+import com.thoughtworks.go.server.newsecurity.filters.VerifyAuthorityFilter;
+import com.thoughtworks.go.server.security.GoAuthority;
+import org.springframework.security.util.FilterChainProxy;
+
+import javax.servlet.Filter;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public class AuthorizeAgentFilterChain extends FilterChainProxy {
+    public AuthorizeAgentFilterChain() {
```

Comment:
This is a replacement for these lines:
https://github.com/gocd/gocd/blob/7672cd956ef5c62f9957592ffc543d11509a31d6/server/webapp/WEB-INF/applicationContext-acegi-security.xml#L198-L215